### DISCUSSION TITLE
silencing commands promise output, clarify inform attribute please?

### DISCUSSION THREAD

**Discussion by craigcomstock** [cfengine/core#5138] [Q&A] [answered]
I am working on this build module to upgrade all packages and noticed that the output was rather verbose. I don't need all of that in my logs.

I thought that `inform => "false"` as described in the docs: https://docs.cfengine.com/docs/master/reference-promise-types-commands.html#inform, would silence output but it seems it does not.

I tried a different contain body, [in_shell_and_silent](https://docs.cfengine.com/docs/master/reference-masterfiles-policy-framework-lib-commands.html#in_shell_and_silent), and that seems to do what I want.

This seems confusing to me. Are the docs for `inform` not quite right? Or am I misunderstanding them somehow?

> Description: Suppress inform output from commands
>
> By default, when running with inform level logging, commands that are executed produce output. This attribute can be used to suppress this output which may not be useful in all cases.

**Comment by nickanderson** [comment, marked as answer]
`inform => "false"` suppresses the info level logs from commands promises. In the example below three commands are run. One suppressing nothing, one suppressing command *output* and one suppressing both command output and *inform* log messages.

```cfengine3
bundle agent __main__
{
  commands:
      "echo"
        arglist => { "Hello", "with", "inform true",
                     "without silence" },
        inform => "true",
        contain => in_shell,
        handle => "i_suppress_nothing";

      "echo"
        arglist => { "Hello", "with", "inform true",
                     "with contain in_shell_and_silent" },
        inform => "true",
        contain => in_shell_and_silent,
        handle => "i_suppress_command_output";

      # NOTE: You won't see this command execution when running cf-agent with info
      # level logging (--log-level info, --inform, -I)

      "echo"
        arglist => { "Hello", "with", 'inform false',
                     "with contain in_shell_and_silent" },
        inform => "false",
        contain => in_shell_and_silent,
        handle => "i_suppress_command_output_and_inform_output";
}
```

    # cf-agent --no-lock --log-level info --file the_difference_between_contain_silent_and_inform_false_in_commands_promises.cf
        info: Executing 'no timeout' ... 'echo Hello with inform true without silence'
      notice: Q: "...echo Hello with": Hello with inform true without silence
        info: Last 1 quoted lines were generated by promiser 'echo Hello with inform true without silence'
        info: Completed execution of 'echo Hello with inform true without silence'
        info: Executing 'no timeout' ... 'echo Hello with inform true with contain in_shell_and_silent'
        info: Completed execution of 'echo Hello with inform true with contain in_shell_and_silent'

I think the docs could probably be improved.

**Reply by craigcomstock** [reply]
Right, I definitely think the docs make it sound like no command output will be seen and from your example that's not the case. Thanks!
